PRASCore/src/Results/GeneratorAvailability.jl

```diff
@@ -0,0 +1,81 @@
+"""
```

Comment:
The structs are distinct enough (at a minimum, different property names) that I don't think this would be an easy win (we'd need to add an abstraction layer anyways). I do like the idea of defining abstract methods on `ResultSpec` to more formally codify the interface, even if they're never going to be invoked - they could just return an error saying the method isn't implemented.